File: table1.py
from IDL2ClassGroup import ICG2, ClassNum, generator

D1 = -10000
D2 = D1-50

for D in range(D1,D2,-1):
    try: ICG2.init(D)
    except: continue
    h = ClassNum()
    G,h1 = generator()
    if h!=h1: print('wrong')
    print(D,h,G)

File: C++/table1.cpp
#include "IDL2ClassGroup.h"
using namespace NTL;

main() {
    long d, d1(-1000), d2(d1-50), i;
    ZZ h;
    Vec<Pair<ICG2, long> > G;
    for(d=d1; d>=d2; d--) {
        try { ICG2::init(d); }
        catch(std::exception) { continue; }
        ICG2::ClassNum(h);
        i = generator(G);
        if(i!=h) Error("i!=h");
        std::cout << ZZ2::D << ' ';
        std::cout << h << ' ';
        std::cout << G << ' ';
        std::cout << std::endl;
    }
}